### PR TITLE
adds commitment merkle tree

### LINKED ISSUE #92: Fix finalise deposits

### PR DESCRIPTION
**Pull request by vaibhavchellani**
FIxes #92

### CODE REVIEW

File: listener/processors.go [hunk 1]
```diff
@@ -98,25 +99,25 @@ func (s *Syncer) processDepositSubtreeCreated(eventName string, abiObject *abi.A
...
+		fmt.Println("Unable to determine")
 		panic(err)
 	}
 
 	if !catchingup {
-		s.SendDepositFinalisationTx()
+		err = s.DBInstance.AttachDepositInfo(event.Root)
+		if err != nil {
+			// TODO do something with this error
+			fmt.Println("Unable to attack deposit information:", err)
```
Comment (ChihChengLiang) [inline listener/processors.go:114]: Nitpick
```suggestion
			fmt.Println("Unable to attach deposit information:", err)
```

File: listener/syncer.go [hunk 2]
```diff
@@ -264,6 +266,7 @@ func (s *Syncer) processEvents(logs []ethTypes.Log, header ethTypes.Header) {
 				case "DepositsFinalised":
 					s.processDepositFinalised(selectedEvent.Name, &abiObject, &vLog)
 				default:
+					fmt.Println("Default here")
```
Comment (ChihChengLiang) [inline listener/syncer.go:269]: What's the difference between using `fmt.Println` and `s.Logger.Debug`?

**Review by ChihChengLiang — APPROVED** [review]
Looks good to me. Add a nitpick

File: listener/processors.go (hunk 1 above)
Comment (vaibhavchellani) [reply listener/processors.go:114]: Damm tabnine

File: listener/syncer.go (hunk 2 above)
Comment (vaibhavchellani) [reply listener/syncer.go:269]: Just that it `s.Logger.Debug` prints the log in the standard of the application whereas `fmt.Println` is just dumped on stdOutput. I intend to remove all `fmt` statements in the next PR's

**Pull request merged**